bindings/C/c/adios2_c_internal.h

```diff
@@ -24,6 +26,8 @@ namespace
 template <int adios2_type>
 struct MapAdios2Type;
 
+inline adios2_error String2CAPI(const std::string &s, char *buf, size_t *size);
+
 } // anonymous namespace
 
 #include "adios2_c_internal.inl"
```

Comment:
there must be a better way to make this private into a cpp file. This is C++ code which (while hidden) that is publicly expose in C bindings after a `make install`. Not sure will help C developers.

Reply:
This isn't specific to this patch: I introduced the adios2_internal.h header because I needed a place to put stuff that's needed by the interface code (which is written in C++), but not part of the C API. I could have put it somewhere in helper/, but since it's specific to the C bindings, that didn't feel right, so this is where it went.

If you have a better place for a header file that is private to bindings but not part of the interface, let me know. FWIW, the python interface does similar things, though there of course it's obvious that none of the bindings/Python/*.h are part of the Python interface ;(

Reply:
@germasch how was this before your PR?

Reply:
I've never changed anything about python, my point is that the bindings/python has header files that are used to compile the python interface code (which is written in C++). (And that obviously header files aren't possible part of a python interface).

For the C bindings there used to be only header files that are installed as part of the API, so they were used both as API and they were included in the C++ code that creates the C bindings. Because they were part of the C API, I wasn't able to add some C++ helper code there, which I needed to build the interface. That's why added (and in this PR slightly extended) the not-installed adios_internal.{h,inl}. It's pat of the binding, but not part of the API. An alternative could be to name it `private/adios2_c_helper.h` or something like that.

Reply:
@germasch Not sure when I implied that Python bindings were changed, my point is that those are never accessed by the Python user, unlike headers in C. See my comment about moving it to helper:: as that has been the team's preference with these kind of functionality. There is no need to return `adios2_error`